src/test/java/io/jenkins/plugins/analysis/core/history/PreviousRunReferenceTest.java

```diff
@@ -384,30 +368,31 @@ void twoFalseAction_withSuccessful_shouldReturnNothing() {
         PreviousRunReference previousRun = new PreviousRunReference(baseline, selector,
                 true);
 
-        Optional<ResultAction> resultAction1 = previousRun.getReferenceAction();
-        Optional<ResultAction> resultAction2 = previousRun.getReferenceAction();
+        when(previousRunMock.getResult()).thenReturn(Result.FAILURE);
+        when(analysis.getOverallResult()).thenReturn(Result.FAILURE);
+        assertThat(previousRun.getReferenceAction()).isEmpty();
+
+        when(previousRunMock.getResult()).thenReturn(Result.FAILURE);
+        when(analysis.getOverallResult()).thenReturn(Result.FAILURE);
+        assertThat(previousRun.getReferenceAction()).isEmpty();
 
```

Comment:
testing two times the same way through the code is useless.
so you can simplify it by just tesing it once.

Don't forget to change the methodname and description.